### PR TITLE
Allow starting in JPMS mode

### PR DESCRIPTION
**Pull request by dmlloyd**
* Use public lookup when looking for public main method
* Add a missing service requirement for `Policy`

For example: `java -p jboss-modules.jar -m org.jboss.modules ...`

More changes are likely going to be needed to get a full startup of a real application.

### CODE REVIEW

**Comment by dmlloyd** [comment]
With this last version, it is possible to start up `wildfly-core` using a JPMS module path which includes `jboss-modules` plus `jboss-logmanager` and its transitive dependency set, and have module resolution working properly.

The versions of `jboss-logmanager` which have a module descriptor are not compatible with `wildfly-core` yet, so there is a limiting factor.

File: src/main/java/module-info.java [hunk 1]
```diff
@@ -17,6 +17,9 @@
  */
 
 module org.jboss.modules {
+  // for consumers of this module
+  requires java.se;
```
Comment (ropalka) [inline src/main/java/module-info.java:21]: When we require java.se we can drop all of the following redundant requirements:
 *  requires java.desktop;
 * requires java.instrument;
 * requires java.logging;
 * requires java.management;
 * requires java.prefs;

**Review by ropalka — CHANGES_REQUESTED** [review]

File: src/main/java/module-info.java (hunk 1 above)
Comment (dmlloyd) [reply src/main/java/module-info.java:21]: Good point.

**Review by ropalka — APPROVED** [review]

**Comment by ropalka** [comment]
Thanks @dmlloyd !

**Pull request merged**